libraries/chain/block_header_state.cpp

```diff
@@ -259,14 +257,20 @@ void finish_next(const block_header_state& prev,
    // -----------------------------------
    next_header_state.block_id = next_header_state.header.calculate_id();
 
-   // Now that we have the block id of the new block, log what changed.
-   // -----------------------------------------------------------------
-   if (next_header_state.active_finalizer_policy != prev.active_finalizer_policy) {
-      const auto& act = next_header_state.active_finalizer_policy;
-      if (log) {
+   if (log) {
+      // Now that we have the block id of the new block, log what changed.
+      // -----------------------------------------------------------------
+      if (f_ext.new_finalizer_policy_diff) {
+         dlog("New finalizer policy proposed in block ${id}: ${pol}",
+              ("id", next_header_state.block_id)("pol", new_finalizer_policy));
+      }
+
+      if (next_header_state.active_finalizer_policy != prev.active_finalizer_policy) {
+         const auto& act = next_header_state.active_finalizer_policy;
          ilog("Finalizer policy generation change: ${old_gen} -> ${new_gen}",
               ("old_gen", prev.active_finalizer_policy->generation)("new_gen",act->generation));
-         ilog("New finalizer policy becoming active in block ${id}: ${pol}",("id", next_header_state.block_id)("pol", *act));
+         ilog("New finalizer policy becoming active in block ${id}: ${pol}",
```

Comment:
Here and above I think it would be nice to include the block number along with the block id.

Reply:
done.